src/scripts/InstanceScripts/Wotlk/Raid_IceCrownCitadel.cpp

```diff
@@ -0,0 +1,2077 @@
+    {
+        getCreature()->Despawn(100, 0);
+        pTarget->RemoveAura(SPELL_IMPALED);
+    }
+
+    void OnDied(Unit* pTarget) override
+    {       
+        if (summon)
+            summon->RemoveAura(SPELL_IMPALED);
+         
+        getCreature()->Despawn(100, 0);
+    }
+
+    void AIUpdate() override
+    {
+        if (!hasTrappedUnit)
+            return;
+
+        scriptEvents.updateEvents(GetAIUpdateFreq(), getScriptPhase());
+
+        if (scriptEvents.getFinishedEvent() == EVENT_FAIL_BONED)
+            if (mInstance)
+                mInstance->setData(DATA_BONED_ACHIEVEMENT, uint32_t(false));
+    }
+
+protected:
+    // Common
+    InstanceScript* mInstance;
+
+    // Summon
+    Unit* summon;
+
+    bool hasTrappedUnit;
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////
+/// Spell: Bone Storm
+class BoneStorm : public SpellScript
+{
+public:
+    void onAuraCreate(Aura* aur) override
+    {
+        // set duration here
+        int32_t duration = 20000;
+        if(aur->GetUnitCaster()->isCreature())
+            duration = static_cast<Creature*>(aur->GetUnitCaster())->GetScript()->RAID_MODE<uint32_t>(20000, 30000, 20000, 30000);
+
+        aur->setOriginalDuration(duration);
+        aur->setMaxDuration(duration);
+        aur->setTimeLeft(duration);
+    }
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////
+/// Spell: Bone Storm Damage
+class BoneStormDamage : public SpellScript
+{
+public:
+    SpellScriptEffectDamage doCalculateEffect(Spell* spell, uint8_t effIndex, int32_t* dmg) override
+    {
+        if (effIndex != EFF_INDEX_0 || spell->GetUnitTarget() == nullptr)
+            return SpellScriptEffectDamage::DAMAGE_DEFAULT;
+
+        auto distance = spell->GetUnitTarget()->GetDistance2dSq(spell->getCaster());
+        // If target is closer than 5 yards, do full damage
+        if (distance <= 5.0f)
+            distance = 1.0f;
+
+        *dmg = float2int32(*dmg / distance);
+        return SpellScriptEffectDamage::DAMAGE_FULL_RECALCULATION;
+    }
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////
+/// Spell: Bone Spike Graveyard
+class BoneSpikeGraveyard : public SpellScript
+{
+    public:
+        virtual void onAuraApply(Aura* aur)
+        {
+            aur->removeAuraEffect(EFF_INDEX_1);
+
+            if (Creature* marrowgar = static_cast<Creature*>(aur->GetUnitCaster()))
+            {
+                CreatureAIScript* marrowgarAI = marrowgar->GetScript();
+                if (!marrowgarAI)
+                    return;
+
+                uint8_t boneSpikeCount = uint8_t(aur->GetUnitCaster()->GetMapMgr()->pInstance->m_difficulty & 1 ? 3 : 1);
+                std::list<Unit*> targets;
+
+                targets.clear();
+
+                uint8_t target = 0;
+                for (target; target < boneSpikeCount; ++target)
```

Comment:
for (uint8_t target = 0; target < boneSpikeCount; ++target)
and remove declaration of target variable one line above